pkg/models/registries/image_type.go

```diff
@@ -0,0 +1,95 @@
+package registries
+
+import (
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+)
+
+type AuthInfo struct {
+	Username   string `json:"username" description:"username"`
+	Password   string `json:"password" description:"password"`
+	ServerHost string `json:"serverhost" description:"registry server host"`
+}
+
+type ImageNameAndSecret struct {
+	ImageName string        `json:"imageName,omitempty" description:"Image full name."`
+	Secret    corev1.Secret `json:"secret,omitempty" description:"Secret holds secret data of a certain type."`
+}
+
+// ImageBlobInfo describes the info of an image.
+type ImageBlobInfo struct {
+	// Status is the status of the image search, such as "failed","succeeded".
+	Status string `json:"status" description:"Status is the status of the image search, such as \"failed\",\"succeeded\"."`
```

Comment:
Why not use http status code to return an error, but use the status field

Reply:
will add error parse